src/oic/utils/keyio.py

```diff
@@ -546,12 +546,15 @@ def get(self, key_use, key_type="", issuer="", kid=None, **kwargs):
                     if kid and key.kid == kid:
                         lst = [key]
                         break
-                    if not key.use or use == key.use:
-                        lst.append(key)
-                        break
+
                     # Verification can be performed by both `sig` and `ver` keys
                     if key_use == 'ver' and key.use in ('sig', 'ver'):
                         lst.append(key)
+                        continue
+
+                    if not key.use or use == key.use:
+                        lst.append(key)
+                        break
```

Comment:
This is not needed as if you would have a key without use, you won't be able to gather any other keys.